service/storage_proxy.cc

```diff
@@ -1811,7 +1811,15 @@ storage_proxy::storage_proxy(distributed<database>& db, storage_proxy::config cf
 }
 
 storage_proxy::unique_response_handler::unique_response_handler(storage_proxy& p_, response_id_type id_) : id(id_), p(p_) {}
-storage_proxy::unique_response_handler::unique_response_handler(unique_response_handler&& x) : id(x.id), p(x.p) { x.id = 0; };
+storage_proxy::unique_response_handler::unique_response_handler(unique_response_handler&& x) noexcept : id(x.id), p(x.p) { x.id = 0; };
+
+storage_proxy::unique_response_handler&
+storage_proxy::unique_response_handler::operator=(unique_response_handler&& x) noexcept {
+    // this->p must equal x.p
```

Comment:
Yes, it probably makes sense that all p's are the same, but why not copy it anyway as we do in the other constructors (maybe you'll need to change p to be a pointer instead of a reference...) - or if you don't want to do this, at least assert() that they are the same?

Reply:
We have only one storage proxy. How could it be different?

Why waste cycles checking something that couldn't be false?

This isn't some generic class used everywhere. It's scoped inside storage_proxy and completely managed by it.

Reply:
Ok, if you're comfortable with it... I was just pointing out that the same class already has other constructors, which do set p - but I agree that setting p the first time (copy constructor) is necessary, but *resetting* it to the same thing (in copy assignment) is a waste. So ok...